generated/osdep/meson.build

```diff
@@ -0,0 +1,68 @@
+xcrun = find_program('xcrun')
+swift_prog = find_program(run_command(xcrun, '-find', 'swift').stdout().strip())
+
+# Split the returned output based on '.'
+# The last character of the first array member is the major version.
+# We don't care about potential patch/point releases so we only fetch the
+# minor version.
+ret = run_command(swift_prog, '-version').stdout().split('.')
+swift_ver = ret[0].substring(-1) + '.' + ret[1].substring(0,1)
+
+if swift_ver.version_compare('<4.1')
+    error('A minimum Swift version of 4.1 is required!')
+elif swift_ver.version_compare('>=6.0')
```

Comment:
why do we fail with swift 6? just because the swift compiler is at version 6 doesn't mean it can't build version 5 only code. just set "-swift-version 5".

https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/waftools/detections/compiler_swift.py#L26-L27

Reply:
My mistake. I misread this.